### PR TITLE
Fix bug where block placeholders are not saved with pull step declared in `deployments` section of a `prefect.yaml` file

### PR DESCRIPTION
**Pull request by desertaxle**
We recently introduced the ability to resolve block document references in the `deployments` section of a `prefect.yaml` file. This introduced a bug where block references used in `pull` actions declared in the `deployments` section of a `prefect.yaml` would be resolved before the `pull` action was saved on the server, so the block document reference did not persist as expected. This PR fixes that by retrieving the `pull` action before block document resolution.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: src/prefect/cli/deploy.py [hunk 1]
```diff
@@ -600,7 +601,7 @@ async def _run_single_deploy(
         triggers = []
 
     pull_steps = (
-        deploy_config.get("pull")
+        pull_steps
         or actions.get("pull")
```
Comment (zzstoatzz) [inline src/prefect/cli/deploy.py:605]: would we still need this line? it looks like thats now on 340 as well

**Review by bunchesofdonald — APPROVED** [review]
Looks good!

File: src/prefect/cli/deploy.py (hunk 1 above)
Comment (zzstoatzz) [reply src/prefect/cli/deploy.py:605]: nevermind, we may have modified `actions`

Comment (desertaxle) [reply src/prefect/cli/deploy.py:605]: Yeah, there are some changes to `actions` that happen when we generate a pull step that need to be available here.

**Review by zzstoatzz — APPROVED** [review]

**Pull request merged**